internal/adminapi/client_test.go

```diff
@@ -0,0 +1,239 @@
+package adminapi
+
+import (
+	"bytes"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/tls"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"github.com/stretchr/testify/assert"
+	"io/ioutil"
+	"math/big"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"strings"
+	"testing"
+	"time"
+)
+
+var ca *x509.Certificate
+var caPEM *bytes.Buffer
+var caPrivateKeyPEM *bytes.Buffer
+var certPEM *bytes.Buffer
+var certPrivateKeyPEM *bytes.Buffer
+
+func TestMakeHTTPClient(t *testing.T) {
+
+	buildTLS(t)
+
+	var opts = HTTPClientOpts{
+		true,
+		"",
+		"",
+		caPEM.String(),
+		nil,
+		"",
+		certPEM.String(),
+		"",
+		certPrivateKeyPEM.String(),
+	}
+
+	httpclient, err := MakeHTTPClient(&opts)
+	if err != nil {
+		t.Errorf("Fail to make the HTTP client - %s", err.Error())
+	}
+
+	assert.NotNil(t, httpclient)
+
+	validate(t, httpclient)
+
+}
+
+func TestMakeHTTPClientWithFilePaths(t *testing.T) {
+
+	buildTLS(t)
+
+	caFile, err := ioutil.TempFile("", "ca.crt")
```

Comment:
Nitpicks: The `ioutil.TempFile()` can be replaced with `os.CreateTemp()` in recent versions of Go, and using `""` as the directory means that the file will be placed in the local code directory which while not problematic in most cases it's common to use the system temporary directory for this purpose so that if something were to go wrong with the code the system can be eventually responsible for cleaning up the temp files:

```suggestion
	caFile, err := os.CreateTemp(os.TempDir(), "ca.crt")
```

Reply:
Fixed